fala/apps/adviser/forms.py

```diff
@@ -69,67 +70,88 @@ def __init__(self, *args, **kwargs):
         kwargs.setdefault("label_suffix", "")
         super(AdviserRootForm, self).__init__(*args, **kwargs)
 
-    def clean(self):
+    def clean(self):  # noqa: max-complexity=13 (increasing the complexity of a function so flake8 is happy)
         data = self.cleaned_data
         postcode = data.get("postcode")
         name = data.get("name")
 
-        # Check if both postcode and name are missing
+        # Add error if both postcode and name are missing
         if not postcode and not name:
             self.add_error("postcode", _("Enter a postcode"))
             self.add_error("name", _("Enter an organisation name"))
             return data
 
-        # Validate postcode if provided and region is either England/Wales or Scotland
-        if postcode and self.region in {Region.ENGLAND_OR_WALES, Region.SCOTLAND}:
-            if not self._valid_postcode(postcode):
+        # Do search on organisation name even if postcode missing
+        if not postcode and name:
+            self._region = Region.ENGLAND_OR_WALES
+            return data
+
+        # Validate postcode if provided and then determine region
+        if postcode:
+            valid_postcode = self.validate_postcode_and_return_country(postcode)
+            if not valid_postcode:
                 self.add_error("postcode", _("Enter a valid postcode"))
+            else:
+                country, nhs_ha = valid_postcode
```

Comment:
Could you add a comment around the use of nhs_ha? It isn't clear from the code why we are using that so would be good to document the reasoning